textalyze.js

```diff
@@ -44,8 +44,8 @@ function itemCounts(array) {
  * @returns {Map} itemFrequencies - A map containing the frequencies of each char based on the total count.
  */
 function itemFrequencies(itemCounts) {
-  if (typeof itemCounts != 'object' || !(itemCounts instanceof Map)) {
-    throw new TypeError('The item frequencies must be a valid map.');
+  if (typeof itemCounts !== 'object' || !(itemCounts instanceof Map)) {
```

Comment:
If `itemCounts` is a `Map` then `typeof itemCounts` will always be `'object'`, so the first check here is redundant.